lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java

```diff
@@ -85,14 +82,36 @@ public APIGatewayProxyResponseEvent handleRequest(
         LogHelper.attachComponentIdToLogs();
 
         try {
-            String ipvSessionId = RequestHelper.getIpvSessionId(input);
+            String ipvSessionId = RequestHelper.getIpvSessionIdAllowNull(input);
+            String clientSessionId = RequestHelper.getClientOAuthSessionId(input);
             String ipAddress = RequestHelper.getIpAddress(input);
-            IpvSessionItem ipvSessionItem = sessionService.getIpvSession(ipvSessionId);
 
-            ClientOAuthSessionItem clientOAuthSessionItem =
-                    clientOAuthSessionService.getClientOAuthSession(
-                            ipvSessionItem.getClientOAuthSessionId());
+            LogHelper.attachIpvSessionIdToLogs(ipvSessionId);
+
+            IpvSessionItem ipvSessionItem = null;
+            ClientOAuthSessionItem clientOAuthSessionItem;
+            if (!StringUtils.isBlank(ipvSessionId)) {
+                ipvSessionItem = sessionService.getIpvSession(ipvSessionId);
+                clientOAuthSessionItem =
+                        clientOAuthSessionService.getClientOAuthSession(
+                                ipvSessionItem.getClientOAuthSessionId());
+            } else if (!StringUtils.isBlank(clientSessionId)) {
+                clientOAuthSessionItem =
+                        clientOAuthSessionService.getClientOAuthSession(clientSessionId);
+                var mapMessage =
+                        new StringMapMessage()
+                                .with("message", "No ipvSession for existing ClientOAuthSession")
+                                .with("clientOAuthSessionId", clientSessionId);
+                LOGGER.info(mapMessage);
+                // We don't have ipvSession here.....should we generate a IPVSession here with this
+                // clientSessionId and then update its auth code below
```

Comment:
Do we want to keep this comment / did this get answered?

Reply:
Removed